examples/lvgldemo/lcddev.c

```diff
@@ -0,0 +1,200 @@
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include "lcddev.h"
+
+#include <nuttx/config.h>
+#include <nuttx/lcd/lcd_dev.h>
+
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef LCDDEV_PATH
+#  define LCDDEV_PATH  "/dev/lcd0"
+#endif
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct lcd_state_s
+{
+  int fd;
+  struct fb_videoinfo_s vinfo;
+  struct lcd_planeinfo_s pinfo;
+  bool rotated;
+};
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct lcd_state_s state;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: lcddev_flush
+ *
+ * Description:
+ *   Flush a buffer to the marked area
+ *
+ * Input Parameters:
+ *   x1      - Left coordinate
+ *   y1      - Top coordinate
+ *   x2      - Right coordinate
+ *   y2      - Bottom coordinate
+ *   color_p - A n array of colors
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void lcddev_flush(struct _disp_drv_t *disp_drv, const lv_area_t *area,
+                         lv_color_t *color_p)
+{
+  int ret;
+  struct lcddev_area_s lcd_area;
+
+  lcd_area.row_start = area->y1;
+  lcd_area.row_end = area->y2;
+  lcd_area.col_start = area->x1;
+  lcd_area.col_end = area->x2;
+
+  lcd_area.data = (uint8_t *)color_p;
+
+  ret = ioctl(state.fd, LCDDEVIO_PUTAREA,
+              (unsigned long)((uintptr_t)&lcd_area));
+
+  if (ret < 0)
+    {
+      int errcode = errno;
+
+      fprintf(stderr, "ERROR: ioctl(LCDDEVIO_PUTAREA) failed: %d\n",
+              errcode);
+      close(state.fd);
```

Comment:
```suggestion
      close(state.fd);
      state.fd = -1;
```
Since the file descriptor is now invalid, it is safer to also reset its value.

Reply:
Although marked as resolved, it has not been addressed.